torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -245,7 +245,7 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
         self.padded_sharded_param_size = padded_sharded_param.size()
         if sharded_param.numel() > 0:
             padded_sharded_param[: sharded_param.size(0)].copy_(sharded_param)
-        if self.offload_to_cpu:
+        if self.offload_to_cpu and not torch.empty(0).is_meta:
```

Comment:
```suggestion
        if self.offload_to_cpu and not padded_sharded_param.is_meta:
```

Reply:
resolving in https://github.com/pytorch/pytorch/pull/126305